internal/provider/kubernetes/controller.go

```diff
@@ -1355,6 +1356,12 @@ func (r *gatewayAPIReconciler) processGateways(ctx context.Context, managedGC *g
 		return err
 	}
 
+	mergedGateways := false
```

Comment:
can we use `r.mergeGateways.Has(gcName)` here?

Reply:
Updated with this